### PR TITLE
OP-97 - New Anamnesis module

### PR DESCRIPTION
**Pull request by goto-eof**
See OP-97.

Checks fail because this PR is linked with https://github.com/informatici/openhospital-core/pull/836.

### CODE REVIEW

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java [hunk 1]
```diff
@@ -0,0 +1,2596 @@
+/**
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:1]: missing copyright block

File: bundle/language_en.properties [hunk 2]
```diff
@@ -1634,3 +1634,103 @@ angal.xmpp.usersinfo.border
 angal.xmpp.wantstosharewithyouthisreport.fmt.msg                                                       = \n*** {0} wants to share with you this report: {1}\n
 angal.xmpp.wouldliketosend.fmt.msg                                                                     = {0} would like to send: \n{1}
 angal.xmpp.youhaverejectedthefiletransfer.txt                                                          = you have rejected the file transfer
+angal.anamnesis.alcool                                                        = Alcool
+angal.anamnesis.alcool.no                                                     = No
+angal.anamnesis.alcool.yes                                                    = Yes
+angal.anamnesis.alvo                                                          = Alvo
+angal.anamnesis.alvo.normal                                                   = Normal
+angal.anamnesis.alvo.other                                                    = Other
+angal.anamnesis.cancel                                                        = Cancel
```
Comment (dbmalkovsky) [inline bundle/language_en.properties:1643]: Some of these new properties already exist; for example for "Cancel" there is `angal.common.cancel.btn` and `angal.common.cancel.btn.key` already defined; they should be reused and not defined again.

In addition the standard properties suffixes should be used:
`.btn` for buttons
`.btn.key` for short cut keys
`.msg` for text used in messages
`.label` for text associated with a label, e.g. `Date From:`
`.tooltip` for tooltip text
`.title` for dialog titles
`.border` for text on grouped items
`.fmt.msg` for messages that includes parameter substitutions
`.col` for text used in column headers for tables

As I have been updating files I've been trying to add the appropriate suffix to the property strings; as you can image it is a large job.

Thanks

Comment (mwithi) [reply bundle/language_en.properties:1643]: yes @dbmalkovsky, the code is a porting from a very old version of OH (v1.6)

I don't remember if we already published somewhere in the dev wiki the list of suffixes... if not I will do it!

Comment (mwithi) [reply bundle/language_en.properties]: I've drafted a Wiki Page for @dbmalkovsky work!
https://openhospital.atlassian.net/wiki/spaces/OH/pages/2203779073/Bundles+translations+files

**Review by mwithi — CHANGES_REQUESTED** [review]
It seems that OPD button in Admisison/Patient window is no longer working, as well as New/Edit OPD ones in the OPD Browser window
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at java.desktop/java.awt.Container.addImpl(Container.java:1117)
	at java.desktop/java.awt.Container.add(Container.java:997)
	at org.isf.opd.gui.OpdEditExtended.getJButtonPanel(OpdEditExtended.java:1685)
...
```

Moreover, upon New Patient creation, clicking on Anamnesis module before saving it causes also NPE
```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at org.isf.patient.gui.PatientInsertExtended.lambda$0(PatientInsertExtended.java:402)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
...
```

**Comment by mwithi** [comment]
> done

NPE are fixed, thanks!

What is missing now is a bundle
image

and default selections for the lower part (Physiological History) should be all "Normal" and "No".

File: bundle/language_en.properties [hunk 3]
```diff
@@ -1635,3 +1635,105 @@ angal.xmpp.usersinfo.border
...
+angal.anamnesis.open.nothingtodeclare.txt                                                              = Nothing to declare
+angal.anamnesis.open.orto.txt                                                                          = Ortoreumatologic
+angal.anamnesis.open.other.txt                                                                         = Other
+angal.anamnesis.open.recentproblems.border                                                             = Recent Problems (Open)
+angal.anamnesis.open.remarks.border                                                                    = Remarks (Open)
+angal.anamnesis.open.respiratory.txt                                                                   = Respiratory
+angal.anamnesis.open.tooltip.excludenothingtodeclare.txt                                               = Exclude 'Nothing to declare'
+angal.anamnesis.open.tooltip.resetrecentproblems.txt                                                   = Reset Recent problems                                                       = Anamnesis
+angal.opd.anamnesis.btn.key                                                                            = M
```
Comment (mwithi) [inline bundle/language_en.properties:1739]: what's wrong here?

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:1]: I don't see the copyright block yet

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java [hunk 4]
```diff
@@ -0,0 +1,2629 @@
...
+	private JCheckBox jCheckBoxPathClosedRespiratory;
+	private JCheckBox jCheckBoxPathClosedCancer;
+	private JCheckBox jCheckBoxPathClosedOrto;
+	private JCheckBox jCheckBoxPathClosedGyno;
+	private JCheckBox jCheckBoxPathClosedOther;
+	private JLabel jLabelPathClosedDiseases;
+	private VoLimitedTextArea jTextAreaPathClosedNote;
+	private JScrollPane jScrollPanePathClosedNote;
+
+//	 PATHOLOGICAL OPEN
```
Comment (mwithi) [inline src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:110]: Fix comment

File: bundle/language_en.properties [hunk 5]
```diff
@@ -1635,3 +1635,105 @@ angal.xmpp.usersinfo.border
...
 angal.xmpp.wouldliketosend.fmt.msg                                                                     = {0} would like to send: \n{1}
 angal.xmpp.youhaverejectedthefiletransfer.txt                                                          = you have rejected the file transfer
+angal.anamnesis.alcool.txt                                                                             = Alcool
+angal.anamnesis.alcool.no.txt                                                                          = No
+angal.anamnesis.alcool.yes.txt                                                                         = Yes
+angal.anamnesis.alvo.txt                                                                               = Alvo
+angal.anamnesis.alvo.normal.txt                                                                        = Normal
+angal.anamnesis.alvo.other.txt                                                                         = Other
+angal.anamnesis.cancel.txt               	                                                           = Cancel
```
Comment (mwithi) [inline bundle/language_en.properties:1644]: alignment

File: bundle/language_en.properties [hunk 6]
```diff
@@ -1635,3 +1635,105 @@ angal.xmpp.usersinfo.border
...
+angal.anamnesis.period.normal.txt                                                                      = Normal
+angal.anamnesis.period.other.txt                                                                       = Other
+angal.anamnesis.physiologicalhistory.border                                                            = Physiological History
+angal.anamnesis.pregnancies.txt                                                                        = Pregnancies
+angal.anamnesis.pregnancies.abort.txt                                                                  = Aborted
+angal.anamnesis.pregnancies.delivery.txt                                                               = Delivered
+angal.anamnesis.pregnancies.no.txt                                                                     = No
+angal.anamnesis.pregnancies.nr.txt                                                                     = Number
+angal.anamnesis.pregnancies.yes.txt                                                                    = Yes
+angal.anamnesis.save.txt          		                                                               = Save
```
Comment (mwithi) [inline bundle/language_en.properties:1712]: alignment

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java [hunk 7]
```diff
@@ -0,0 +1,2629 @@
...
+//			updateGUIHistory();
+//		}
+//	}
+//
+//	private Action getActionExcludePathOpenNothing() {
+//		if (actionExcludePathOpenNothing == null) {
+//			actionExcludePathOpenNothing = new SwingActionExcludePathOpenNothing();
+//		}
+//		return actionExcludePathOpenNothing;
+//	}
```
Comment (mwithi) [inline src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:2374]: remove this part

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java [hunk 8]
```diff
@@ -0,0 +1,2629 @@
...
+//				patex = exaServ.getDefaultPatientExamination(pat.getCode());
+//			}
+//
+//			GenderPatientExamination gPatex = new GenderPatientExamination(patex, pat.getSex() == 'M');
+//			PatientExaminationEdit dialog = new PatientExaminationEdit(PatientHistoryEdit.this, gPatex);
+//			dialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
+//			dialog.pack();
+//			dialog.setLocationRelativeTo(null);
+//			dialog.setVisible(true);
+		}
```
Comment (mwithi) [inline src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:2618]: So this action does nothing?

File: src/main/java/org/isf/patient/gui/PatientInsert.java [hunk 9]
```diff
@@ -238,11 +238,13 @@ private JPanel getJButtonPanel() {
 		if (jButtonPanel == null) {
 			jButtonPanel = new JPanel();
 			jButtonPanel.add(getJOkButton(), null);
-			jButtonPanel.add(getJCancelButton(), null);
+					jButtonPanel.add(getJCancelButton(), null);
```
Comment (mwithi) [inline src/main/java/org/isf/patient/gui/PatientInsert.java:241]: alignment

File: src/main/java/org/isf/patient/gui/PatientInsertExtended.java [hunk 10]
```diff
@@ -79,27 +85,24 @@
...
- * 						   - modified age field from int to varchar
- * 28/08/2008 - cla		   - added tooltip for age field and checking name and age for patient editing
- * 05/09/2008 - alex       - added patient code
- * 01/01/2009 - Fabrizio   - modified assignment to age field to set an int value
- * ------------------------------------------
+ * ------------------------------------------ PatientInsertExtended - model for the patient entry ----------------------------------------- modification history
+ * 11/08/2008 - alessandro - added mother and father names textfield 11/08/2008 - alessandro - changed economicStatut -> hasInsurance 19/08/2008 - mex - changed
+ * educational level with blood type 26/08/2008 - cla - added calendar for calculating age - modified age field from int to varchar 28/08/2008 - cla - added
+ * tooltip for age field and checking name and age for patient editing 05/09/2008 - alex - added patient code 01/01/2009 - Fabrizio - modified assignment to age
+ * field to set an int value ------------------------------------------
```
Comment (mwithi) [inline src/main/java/org/isf/patient/gui/PatientInsertExtended.java:92]: revert this formatting

File: bundle/language_en.properties (hunk 3 above)
Comment (dbmalkovsky) [reply bundle/language_en.properties:1739]: Line 1738 has two equal signs; `= Reset Recent problems` and further to the right ` = Anamnesis`.

BTW, the tooltip text should not capitalize `Recent`; that is `Reset Recent problems` -> `Reset recent problems`

**Review by mwithi — APPROVED** [review]

File: src/main/java/org/isf/opd/gui/OpdEditExtended.java [hunk 11]
```diff
@@ -567,7 +578,7 @@ private void initialize() {
 		this.setContentPane(getMainPanel());
 		pack();
 		setMinimumSize(this.getSize());
-		setLocationRelativeTo(null);
```
Comment (mwithi) [inline src/main/java/org/isf/opd/gui/OpdEditExtended.java:570]: please revert this deletion

setLocationRelativeTo(null); makes the window to appear screen centered.

**Review by mwithi — CHANGES_REQUESTED** [review]
A part some changes (see comments) the previous problems have been fixed.

File: src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java (hunk 8 above)
Comment (goto-eof) [reply src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java:2618]: obsolete code. I removed also another piece of code. Please make a fast check of the GUI. I have made some tests and seems that it is all ok.

**Review by mwithi — APPROVED** [review]

**Pull request merged**